### DISCUSSION TITLE
AttributeError: module 'platform' has no attribute 'machine'

### DISCUSSION THREAD

**Discussion by yunshangrourou** [sqlalchemy/sqlalchemy#7222] [Usage Questions]
### Describe the bug

I use Pycharm to execute code remotely on the Raspberry Pi 4b. I would like to use the sum function of func, so I import func into the code.

```python
from sqlalchemy import func
```

But when I execute it, I get an error

### To Reproduce

```python
import datetime
import sys
from PyQt5.QtWidgets import *
from PyQt5.QtCore import pyqtSignal, Qt
from sqlalchemy import func

...
```

### Error

```
ssh://pi@192.168.0.111:22/usr/bin/python3 -u /data/test_table.py
Traceback (most recent call last):
  File "/data/test_table.py", line 5, in <module>
    from sqlalchemy import func
  File "/home/pi/.local/lib/python3.7/site-packages/sqlalchemy/__init__.py", line 8, in <module>
    from . import util as _util  # noqa
  File "/home/pi/.local/lib/python3.7/site-packages/sqlalchemy/util/__init__.py", line 14, in <module>
    from ._collections import coerce_generator_arg  # noqa
  File "/home/pi/.local/lib/python3.7/site-packages/sqlalchemy/util/_collections.py", line 16, in <module>
    from .compat import binary_types
  File "/home/pi/.local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 31, in <module>
    arm = "aarch" in platform.machine().lower()
AttributeError: module 'platform' has no attribute 'machine'

Process finished with exit code 1

``` 

### Versions

- OS: Raspbian GNU/Linux 10
- Python: 3.7.3
- SQLAlchemy: 1.3.23
- Database: sqlite3
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): sqlite3

**Comment by CaselIT** [comment]
Hi,

I think your python install is broken. Please ensure that it's properly installed.
`platform.machine` is a method in the standard lib, that is available in python 3.7: https://docs.python.org/3.7/library/platform.html

SQLAlchemy assumes the standard lib is available, otherwise we would need to replicate everything

**Comment by zzzeek** [comment]
hi -

let me note that a common reason that an error like "`<module> has no attribute <x>`", when "`<module>`" is a python standard library name like "platform" is overwritten by some other package or module in the current library.   so you want to look in your application as well as in your site-packages to see if anything is overwriting it.

A good way just to check the install is to see what `__file__` has to say, like this:

```
$ python
Python 3.9.7 (default, Aug 30 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.__file__
'/usr/lib64/python3.9/platform.py'
```
